From: Dylan Baker <dylan@pnwbakers.com>
Date: Tue, 4 Dec 2018 13:52:21 -0800
Subject: [5/5] meson: Add toggle for glx-direct

From patchwork Tue Dec  4 21:52:21 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Id: 265800
Message-Id: <20181204215221.18155-6-dylan@pnwbakers.com>
To: mesa-dev@lists.freedesktop.org
Cc: ignatenko.brain@gmail.com, cefiar@gmail.com

GNU Hurd needs to turn off glx-direct, rather than special case it,
we'll just add a toggle.
---
 meson.build       | 4 +---
 meson_options.txt | 6 ++++++
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index 27d15f6..3d38c20 100644
--- a/meson.build
+++ b/meson.build
@@ -52,6 +52,7 @@ with_valgrind = get_option('valgrind')
 with_libunwind = get_option('libunwind')
 with_asm = get_option('asm')
 with_glx_read_only_text = get_option('glx-read-only-text')
+with_glx_direct = get_option('glx-direct')
 with_osmesa = get_option('osmesa')
 with_swr_arches = get_option('swr-arches')
 with_tools = get_option('tools')
@@ -366,9 +367,6 @@ if with_glvnd
   endif
 endif
 
-# TODO: toggle for this
-with_glx_direct = true
-
 if with_vulkan_icd_dir == ''
   with_vulkan_icd_dir = join_paths(get_option('datadir'), 'vulkan/icd.d')
 endif
diff --git a/meson_options.txt b/meson_options.txt
index a1d5ab0..589d10b 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -318,3 +318,9 @@ option(
   choices : ['auto', 'true', 'false'],
   description : 'Enable VK_EXT_acquire_xlib_display.'
 )
+option(
+  'glx-direct',
+  type : 'boolean',
+  value : true,
+  description : 'Enable direct rendering in GLX and EGL for DRI',
+)
